### PR TITLE
WIP: Ability to insert a <disp-quote> element.

### PR DESCRIPTION
**Pull request by michael**

### CODE REVIEW

**Comment by michael** [comment]
@NateWr won't get to this today. Hopefully tomorrow.

**Comment by michael** [comment]
Insertion works fine now! However, performing an undo right after insertion causes an error. Looks like the XRefComponent gets rerendered while the node has actually been deleted already. I figure the XRefComponent has not been disposed properly (unregister from labels:generated event)?

**Comment by michael** [comment]
Stacktrace: 
image

**Comment by michael** [comment]
@oliver---- I implemented InsertDispQuoteCommand. However, the validator claims that `<disp-quote-content>` is not allowed inside `<disp-quote>` and rejects the operation, while this should be perfectly valid. 

To reproduce the error: Click Insert -> Blockquote

The problem is that `schema.findLastValidPos(this, child.type)` returns -1 instead of 0. Might be an edge case that the DFA misses.

image

**Comment by oliver----** [comment]
There is no problem with dispose. But what you do there is to emit labels:generated during the 'update' stage, where XRefComponent is calling rerender right away.
Thus, rerender is called too earlier, particularly before the TextProperty has been rerendered.

**Comment by oliver----** [comment]
The problem with the insert is, that it chokes with the partially invalid element (`<attrib> is not there yet, but mandatory)
I need to rethink this, as we only need to make sure things are valid at the end of a transaction.

File: src/article/TextureEditingInterface.js [hunk 1]
```diff
@@ -17,21 +17,4 @@ export default class TextureEditingInterface extends EditingInterface {
     return this.getDocument().createElement(...args)
   }
 
-  /*
-    2.0 API suggestion (pass only id, not data)
-  */
-  insertBlockNode(nodeId) {
```
Comment (oliver----) [inline src/article/TextureEditingInterface.js]: @michael I think we want to go for a purely Node/Element oriented interface. Using the `id` I consider more low-level and thus more 1.0 style.

Comment (michael) [inline src/article/TextureEditingInterface.js]: You are right. I like that API!

**Comment by oliver----** [comment]
We are now validating changed elements after a transaction.

**Comment by oliver----** [comment]
@michael u can continue

**Comment by michael** [comment]
That looks great. Thanks! Think we are done with this PR. Are we good to publish new Substance preview and merge?

**Comment by oliver----** [comment]
Probably I need to improve the XML API further when working on Stencila, but I understand that we want to get this here out for further testing.

**Comment by michael** [comment]
Should we continue with beta7-preview.x or start with 1.0.0-preview1? :)

**Comment by oliver----** [comment]
let's do it ;)

**Comment by michael** [comment]
@oliver---- i will do it if ok?

**Pull request merged**